core/domain/story_domain.py

```diff
@@ -317,82 +337,72 @@ def validate(self):
                     raise utils.ValidationError(
                         'Expected all destination nodes to exist')
             node.validate()
+            if node.id == self.starting_node_id:
+                starting_node_present = True
+
+        if not starting_node_present:
+            raise utils.ValidationError('Expected starting node to exist.')
+
+        # nodes_queue stores the pending nodes to visit in the story that are
+        # unlocked, in a 'queue' form with a First In First Out structure.
+        nodes_queue = []
+        is_node_visited = [False] * len(self.nodes)
+        starting_node_index = self.find_node_index(self.starting_node_id)
+        nodes_queue.append(self.nodes[starting_node_index].id)
+
+        # The user is assumed to have all the prerequisite skills of the
+        # starting node before starting the story. Also, this list models the
+        # skill IDs acquired by a learner as they progress through the story.
+        simulated_skill_ids = self.nodes[
+            starting_node_index].prerequisite_skill_ids
+
+        # The following loop employs a Breadth First Search from the given
+        # starting node and makes sure that the user has acquired all the
+        # prerequisite skills required by the destination nodes 'unlocked' by
+        # visiting a particular node by the time that node is finished.
+        while len(nodes_queue) > 0:
+            current_node_id = nodes_queue.pop()
+            current_node_index = self.find_node_index(current_node_id)
+            # The following condition checks whether the destination node for
```

Comment:
I don't see a "check" here? It just sets it?

Reply:
Oh, that comment is misplaced, moved the check below, will move this,

Reply:
Done.